include/Gaffer/TweakPlug.inl

```diff
@@ -91,7 +91,12 @@ bool TweakPlug::applyTweak(
 		mode == Gaffer::TweakPlug::Replace ||
 		mode == Gaffer::TweakPlug::Add ||
 		mode == Gaffer::TweakPlug::Subtract ||
-		mode == Gaffer::TweakPlug::Multiply
+		mode == Gaffer::TweakPlug::Multiply ||
+		mode == Gaffer::TweakPlug::Min ||
+		mode == Gaffer::TweakPlug::Max ||
+		mode == Gaffer::TweakPlug::ListAppend ||
+		mode == Gaffer::TweakPlug::ListPrepend ||
+		mode == Gaffer::TweakPlug::ListRemove
```

Comment:
This is the logical extension of the original code, but I think it suggests that the original code was structured in a slightly non-obvious and harder-to-maintain way. I think this whole conditional boils down to `mode != Create`, because in Create mode we don't need to get the `currentValue`. So I think this might be expressed a bit more obviously like so :

```
if( mode == Create )
{
    // Early out. No need to consider any other logic.
    return setDataFunctor( name, newData );
}

// Now deal with the cases that do require a pre-existing value.
const IECore::Data *currentValue = getDataFunctor( name );
....

```

Reply:
Related to this, I'd like to propose an extension to the original specification we had for ListAppend and ListPrepend : if there is no current value, then fall back to `Create` mode. In other words, consider a missing input to be an empty list. My motivation for this is the "pass editor" stuff we've been talking about, because creating a pass will involve doing a ListAppend onto the current list of passes. I don't want to have to do a bunch of `AttributeQuery.exists` juggling to switch between Create mode and ListAppend mode when creating the first pass vs additional passes.

This probably merits some debate, because the original TweakPlug design wasn't this laissez-faire - we could have treated missing inputs as being `0.0` for the numeric cases for instance. Even `Replace` mode is pedantic enough to refuse to create something that didn't exist before. These are probably my main arguments :

- The roots of TweakPlug are in ShaderTweaks, where we wanted to make sure people only tweaked valid shader parameters, and could be sure that all valid parameters were present in the incoming data. Now that TweakPlug has been generalised for use elsewhere, this is less reasonable.
- Whenever we are pedantic about erroring when we could carry on, we find crazy graphs where people work around the error (like my AttributeQuery.exists thing above).
- Usability is important.
- It may be harder to defend `0.0` as the default for a missing number than `[]` as the default for a missing list. I can think of lots of numeric attributes with various defaults, but I can't bring to mind any list attributes (empty or non-empty by default).

Arguments against might be :

- `AttributeTweaks.ignoreMissing` says that tweaks targeting missing attributes will be ignored. Perhaps this could be reworded to "tweaks _requiring_ missing attributes"?
- If you think the input exists, you might appreciate an error informing you that it doesn't. In some circumstances this is undoubtedly true. But experience in production shows that more useful is to build simple graphs that can deal with whatever input is thrown at them.
- If we want this sort of behaviour, we should introduce a more nuanced set of options for `ignoreMissing` - "Error", "Ignore", "UseASensibleDefault". I would argue against that as additional complexity that is hard to explain and understand, and doesn't add much, because I believe what people want is for it to "just work" most of the time.

OK, that's me done debating myself. @murraystevenson, @danieldresser-ie, would be good to get your input on this too.

Reply:
> So I think this might be expressed a bit more obviously like so :

Yes, much cleaner. Fixed in faf26d92a1f944a4b4673f03ddbc489f94e36159